pkg/kv-cache/indexer.go

```diff
@@ -98,6 +98,11 @@ func (k *Indexer) Run(ctx context.Context) {
 	k.tokenizersPool.Run(ctx)
 }
 
+// KVBlockIndex returns the kvblock.Index used by the Indexer.
+func (k *Indexer) KVBlockIndex() kvblock.Index {
```

Comment:
Q: what is the rationale for exposing an internal (?) implementation field?
Can you be assured that the caller (who gets hold of the `kvbloc.Index` won't be holding unto it or mutating it in a way that is not desirable or expected by the `Indexer`?

Reply:
There were two options here:
1. The `kvblock.Index` is initialized before the `kvcache.Indexer` and is passed to the latter on initialization (and thus the first is configured separately)
2. The `kvcache.Indexer` is configured first, and its `kvblock.Index` is extractable to be managed. The latter declares concurrent-access safety so it is fine

The 2nd seemed more sensible but that might be biased by the continuity of development.